superset-frontend/src/embedded/index.tsx

```diff
@@ -51,11 +52,27 @@ const LazyDashboardPage = lazy(
     ),
 );
 
+const EmbededLazyDashboardPage = () => {
+  const uiConfig = useUiConfig();
+
+  // Emit data mask changes to the parent window
+  if (uiConfig?.emitDataMasks) {
+    log('setting up Switchboard event emitter');
+
+    store.subscribe(() => {
+      const state = store.getState();
+      Switchboard.emit('getDataMasks', state.dataMask);
```

Comment:
This will emit `getDataMask` on every change in the store, resulting in many redundant calls. Can we emit only when the dataMask changes, like this:
```
if (uiConfig?.emitDataMasks) {
    log('setting up Switchboard event emitter');

    let previousDataMask = store.getState().dataMask;
    
    store.subscribe(() => {
      const currentState = store.getState();
      const currentDataMask = currentState.dataMask;
      
      // Only emit if the dataMask has changed
      if (previousDataMask !== currentDataMask) {
        Switchboard.emit('getDataMasks', currentDataMask);
        previousDataMask = currentDataMask;
      }
    });
  }
  ```